src/components/cpu/cpu-count/cpu-count.js

```diff
@@ -18,19 +18,35 @@ export class CpuCount extends HTMLElement {
      * Call back for when the component is attached to the DOM
      */
     connectedCallback() {
+        this.defaults = {};
         var defaults = new Defaults();
-        var margin = this.dataset.margin || defaults.margin;
-        var height = (this.dataset.height || defaults.height) - margin.top - margin.bottom;
-        var width = (this.dataset.width || defaults.width) - margin.left - margin.right;
-        var lineColor = this.dataset.lineColor || defaults.lineColor;
-        var unit = (this.dataset.unit || 'count');
-
-        this.innerHTML = "<line-graph data-margin=" + JSON.stringify(margin) +
-            " data-height=" + height +
-            " data-width=" + width +
-            " data-graph=" + this.dataset.graph +
-            " data-line-color=" + lineColor +
-            " data-unit=" + unit + "></lineGraph>";
+        this.defaults.margin = this.dataset.margin || defaults.margin;
+        this.defaults.height = (this.dataset.height || defaults.height) - this.defaults.margin.top - this.defaults.margin.bottom;
+        this.defaults.width = (this.dataset.width || defaults.width) - this.defaults.margin.left - this.defaults.margin.right;
+        this.defaults.lineColor = this.dataset.lineColor || defaults.lineColor;
+
+        this.render();
+    }
+
+    /**
+     * @name render
+     * @description
+     * Kicks off the render process after attribute value has been set & connectedcallback has run.
+     * @param {string} data this param is collected from the data-graph attribute
+     */
+    render () {
+        if (this.graphData && this.defaults) {
+          this.innerHTML = "<line-graph data-margin=" + JSON.stringify(this.defaults.margin) + " data-height=" + this.defaults.height + " data-width=" + this.defaults.width + " data-graph=" + this.graphData + " data-line-color=" + this.defaults.lineColor + "></lineGraph>";
```

Comment:
can we break this down to multi lines so that the line itself is not so long.  Code style standards usually call for lines to not be longer than 80 chars.

https://nickjanetakis.com/blog/80-characters-per-line-is-a-standard-worth-sticking-to-even-today